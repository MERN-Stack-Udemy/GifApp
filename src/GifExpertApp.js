import React, { useState } from 'react'
import { AddCategorie } from './components/AddCategorie';
import { GifsGrid } from './components/GifsGrid';

const GifExpertApp = ( {defaultCategories = []} ) => {

  const [categories, setCategories] = useState( defaultCategories )
  
  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategorie 
        setCategories = { setCategories }
      />

      <hr/>

      <ol>{
        categories.map( ( categorie ) => (
          // return <li key={categorie}> {categorie}</li>
          <GifsGrid 
            key={categorie}
            // class
            category= { categorie }
          />
        ))
      }</ol>
      
    </>
  )

}
export default GifExpertApp;
