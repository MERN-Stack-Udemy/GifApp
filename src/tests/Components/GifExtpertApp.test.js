import { shallow } from 'enzyme'
import GifExpertApp from '../../GifExpertApp'

describe(' testing <GifExpertApp/> component', () => {
  
  test('should make match with de snapshot', () => {
    const wrapper = shallow( <GifExpertApp /> );
    expect( wrapper ).toMatchSnapshot();
  })
  
  test('should display a list of categories ', () => {
    const categories = ['One Punch', 'Dragon Ball'];
    const wrapper = shallow( <GifExpertApp defaultCategories={categories} /> );

    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find('GifsGrid').length ).toBe( categories.length );
  })
  
})
