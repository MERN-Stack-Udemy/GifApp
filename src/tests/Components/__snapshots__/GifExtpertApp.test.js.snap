// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` testing <GifExpertApp/> component should display a list of categories  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategorie
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifsGrid
      category="One Punch"
      key="One Punch"
    />
    <GifsGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[` testing <GifExpertApp/> component should make match with de snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategorie
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
